File: 1-H/java/src/Card.java
public class Card {

    private final String suit;
    private final int point;

    public enum Suit {
        CLUBS("C"), 
        DIAMONDS("D"), 
        HEARTS("H"), 
        SPADES("S");

        private final String representation;

        Suit(String representation) {
            this.representation = representation;
        }

        @Override
        public String toString() {
            return representation;
        }
    }

    public enum Rank {
        R2("2"), 
        R3("3"), 
        R4("4"), 
        R5("5"), 
        R6("6"), 
        R7("7"), 
        R8("8"), 
        R9("9"), 
        R10("10"), 
        J("J"), 
        Q("Q"), 
        K("K"), 
        A("A");

        private final String representation;

        Rank(String representation) {
            this.representation = representation;
        }

        @Override
        public String toString() {
            return representation;
        }
    }

    public Card(String suit, int point) {
        this.suit = suit;
        this.point = point;
    }

    public int getCardPoint() {
        return point;
    }
    
}

File: 1-H/Python/card.py
from enum import Enum, auto

class Suit(Enum):
    CLUB = auto()
    DIAMOND = auto()
    HEART = auto()
    SPADE = auto()

    def __str__(self):
        return self.name

class Rank(Enum):
    R2 = auto()
    R3 = auto()
    R4 = auto()
    R5 = auto()
    R6 = auto()
    R7 = auto()
    R8 = auto()
    R9 = auto()
    R10 = auto()
    J = auto()
    Q = auto()
    K = auto()
    A = auto()

    def __str__(self):
        return self.name

class Card:
    def __init__(self, rank, suit):
        self.suit = suit
        self.rank = rank

    def __str__(self):
        return "{} of {}".format(self.rank, self.suit)

    def __lt__(self, other):
        if self.rank == other.rank:
            return self.suit.value < other.suit.value
        return self.rank.value > other.rank.value

    @property
    def getSuits(self):
        return self.suit
    
    @property
    def getRanks(self):
        return self.rank